PyriteEditor/src/CornellBoxScene.h

```diff
@@ -0,0 +1,172 @@
+            m_ggxShader->bindTexture(brdfLUT, "brdfLUT");
+            m_ggxShader->bindCubemap(*m_irradianceMap, "irrandiance_map");
+            m_ggxShader->bindCubemap(*specularCubemap, "prefilterMap");
+#pragma region RDG
+            m_camera.setPosition(vec3{ 0,0,0 });
+            m_camera.lookAt(vec3{ 0,0,0.f });
+            m_camera.setProjection(pyr::PerspectiveProjection{});
+            m_camController.setCamera(&m_camera);
+            m_ggxShader->addBinding({ .label = "CameraBuffer",   .bufferRef = pcameraBuffer });
+            m_RDG.addPass(&m_depthPrePass);
+            m_RDG.addPass(&m_SSAOPass);
+            m_RDG.addPass(&m_forwardPass);
+            m_RDG.getResourcesManager().addProduced(&m_depthPrePass, "depthBuffer");
+            m_RDG.getResourcesManager().addProduced(&m_SSAOPass, "ssaoTexture_blurred");
+            m_RDG.getResourcesManager().addProduced(&m_SSAOPass, "ssaoTexture");
+            m_RDG.getResourcesManager().addRequirement(&m_SSAOPass, "depthBuffer");
+            m_RDG.getResourcesManager().linkResource(&m_depthPrePass, "depthBuffer", &m_SSAOPass);
+            m_RDG.getResourcesManager().linkResource(&m_SSAOPass, "ssaoTexture_blurred", &m_forwardPass);
+            m_forwardPass.boundCamera = &m_camera;
+            bool bIsGraphValid = m_RDG.getResourcesManager().checkResourcesValidity();
+#pragma endregion RDG
+
+            sceneMeshes.reserve(m_cornellBoxModels.size());
+            for (const std::shared_ptr<pyr::Model>& model : m_cornellBoxModels)
+            {
+                sceneMeshes.emplace_back(pyr::StaticMesh{ model });
+                sceneMeshes.back().getTransform().scale = { 10,10,10 };
+                m_forwardPass.addMeshToPass(&sceneMeshes.back());
+                m_depthPrePass.addMeshToPass(&sceneMeshes.back());
+            }
+            m_lights.Points.push_back({});
+            m_lights.Points.push_back({});
+            m_lights.Directionals.push_back(pyr::DirectionalLight{});
+            m_lights.Spots.push_back({});
+            m_lights.Spots.push_back({});
+            m_lights.Spots.push_back({});
+
+            std::vector<pyr::hlsl_GenericLight> gpu_lights = m_lights.ConvertCollectionToHLSL();
+
+        }
+
+        /// ------------------------------------------------------------------------------------------------------------------------------- ///
+        
+        void update(float delta) override
+        {
+            m_camController.processUserInputs(delta);
+            pcameraBuffer->setData(CameraBuffer::data_t{
+                .mvp = m_camera.getViewProjectionMatrix(),
+                .pos = m_camera.getPosition()
+                });
+            pinvCameBuffer->setData(InverseCameraBuffer::data_t{
+                .inverseViewProj = m_camera.getViewProjectionMatrix().Invert(),
+                .inverseProj = m_camera.getProjectionMatrix().Invert(),
+                .Proj = m_camera.getProjectionMatrix()
+                });
```

Comment:
Shouldn't `setData` calls be in `render()`? To differenciate engine/graphic tasks

Reply:
how about a PreRender() function